### ISSUE TITLE
Allow exceptions to window swallowing

### ISSUE THREAD

**Issue by mrkajetanp** [hyprwm/Hyprland#2002]
As far as I can see, under the current implementation window swallowing is pretty much all or nothing - if I add the terminal in the regex, it will swallow any window it spawns. While that's the desired behaviour almost all the time there's occasionally some exceptions (like `wev`) where you want to keep the default behaviour. Could we add something like `swallow_exception_regex` for this use case?

**Comment by vaxerski** [comment]
how would you even except that to work?? Can't you just make a regex that does _not_ target that one thing that you don't want to swallow?

**Comment by grappas** [comment]
I think swallow_regex =^(classes)$ but in reverse (not terminals but child windows like wev for example)

**Comment by tchofy** [comment]
Most terminals put the working application on their title, or can be configured to do so, so what about excluding by title?
I'm not particularly good with regex yet, but I think it'd go something like: swallow_regex=class:kitty,title:^(?!wev)$

**Comment by mrkajetanp** [comment]
> I think swallow_regex =^(classes)$ but in reverse (not terminals but child windows like wev for example)

Yes exactly, other WMs that do swallowing like DWM for instance already have this functionality. Implementation-wise it just boils down to matching against the regex before a window is about to be swallowed to see if it's one of the exceptions or not.

I implemented a PR that does just that, it's the one mentioned right above this comment. I tested it on my machine and now it works as intended.
